packages/react/src/components/form/form.tsx

```diff
@@ -0,0 +1,195 @@
+  /**
+   * The optional indicator to use for each field.
+   */
+  optionalIndicator?: ReactNode
+  /**
+   * An object managing the read only state of each field.
+   */
+  readOnly?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * An object managing the replace state of each field.
+   */
+  replace?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * An object managing the required state of each field.
+   */
+  required?: boolean | { [key: string]: boolean | undefined }
+  /**
+   * The required indicator to use for each field.
+   */
+  requiredIndicator?: ReactNode
+}
+
+const {
+  PropsContext: FormPropsContext,
+  usePropsContext: useFormPropsContext,
+  withContext,
+  withProvider,
+} = createSlotComponent<FormRootProps, FormStyle, FormContext>(
+  "form",
+  formStyle,
+)
+
+export { FormPropsContext, useFormPropsContext }
+
+/**
+ * `Form` is a component used to group multiple form elements.
+ *
+ * @see https://yamada-ui.com/docs/components/form
+ */
+export const FormRoot = withProvider<"form", FormRootProps>(
+  ({
+    id,
+    children,
+    disabled,
+    errorMessage,
+    helperMessage,
+    invalid,
+    noValidate = true,
+    optionalIndicator,
+    readOnly,
+    replace,
+    required,
+    requiredIndicator,
+    ...rest
+  }) => {
+    const uuid = useId()
+
+    id ??= uuid
+
+    const context = useMemo(
+      () => ({
+        id,
+        disabled,
+        errorMessage,
+        helperMessage,
+        invalid,
+        optionalIndicator,
+        readOnly,
+        replace,
+        required,
+        requiredIndicator,
+      }),
+      [
+        id,
+        disabled,
+        errorMessage,
+        helperMessage,
+        invalid,
+        readOnly,
+        replace,
+        required,
+        optionalIndicator,
+        requiredIndicator,
+      ],
+    )
+
+    return (
+      <FormContext value={context}>
+        <styled.form id={id} noValidate={noValidate} {...rest}>
+          {children}
+        </styled.form>
+      </FormContext>
+    )
+  },
+  "root",
+)()
+
+export interface FormHeaderProps extends HTMLStyledProps {}
+
+export const FormHeader = withContext<"div", FormHeaderProps>("div", "header")()
+
+export interface FormTitleProps extends HTMLStyledProps<"h3"> {}
+
+export const FormTitle = withContext<"h3", FormTitleProps>("h3", "title")()
+
+export interface FormDescriptionProps extends HTMLStyledProps<"p"> {}
+
+export const FormDescription = withContext<"p", FormDescriptionProps>(
+  "p",
+  "description",
+)()
+
+export interface FormBodyProps extends HTMLStyledProps {}
+
+export const FormBody = withContext<"div", FormBodyProps>("div", "body")()
+
+export interface FormGroupProps extends HTMLStyledProps {}
+
+export const FormGroup = withContext<"div", FormGroupProps>("div", "group")()
+
+export interface FormGroupTitleProps extends HTMLStyledProps<"h4"> {}
+
+export const FormGroupTitle = withContext<"h4", FormGroupTitleProps>(
+  "h4",
+  "groupTitle",
+)()
```

Comment:
Shouldn't we use more semantic tags, like we are doing with the Card component?

```suggestion
export interface FormGroupProps extends HTMLStyledProps<"fieldset"> {}

export const FormGroup = withContext<"fieldset", FormGroupProps>("fieldset", "group")()

export interface FormGroupTitleProps extends HTMLStyledProps<"legend"> {}

export const FormGroupTitle = withContext<"legend", FormGroupTitleProps>(
  "legend",
  "groupTitle",
)()
```